cmd/podman/system/events.go

```diff
@@ -162,15 +163,21 @@ func eventsCmd(cmd *cobra.Command, _ []string) error {
 	}
 
 	go func() {
-		err := registry.ContainerEngine().Events(context.Background(), eventOptions)
-		errChannel <- err
+		errChannel <- registry.ContainerEngine().Events(context.Background(), eventOptions)
 	}()
 
 	for {
 		select {
 		case event, ok := <-eventChannel:
 			if !ok {
 				// channel was closed we can exit
+				select {
+				case err := <-errChannel:
+					if err != nil {
+						return err
+					}
+				case <-time.After(1 * time.Millisecond):
```

Comment:
assuming you did this to get a non blocking call on the error channel you should use the `default` case instead

Reply:
Either one assumes that the error will be ready by the time the `select` statement looks at the channel, but yeah, `default` would be cleaner.  Changed it.